src/app/page.tsx

```diff
@@ -1,91 +1,97 @@
 "use client";
 
 import { useEffect, useState } from "react";
+import SearchBar from "@/app/components/SearchBar";
+import SearchStatus from "@/app/components/SearchStatus";
+import AdvocatesTable from "@/app/components/AdvocatesTable";
+import { Advocate } from "@/app/types";
 
 export default function Home() {
-  const [advocates, setAdvocates] = useState([]);
-  const [filteredAdvocates, setFilteredAdvocates] = useState([]);
+  const [advocates, setAdvocates] = useState<Advocate[]>([]);
+  const [filteredAdvocates, setFilteredAdvocates] = useState<Advocate[]>([]);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [error, setError] = useState("");
+  const [loading, setLoading] = useState(true);
 
   useEffect(() => {
     console.log("fetching advocates...");
-    fetch("/api/advocates").then((response) => {
-      response.json().then((jsonResponse) => {
-        setAdvocates(jsonResponse.data);
-        setFilteredAdvocates(jsonResponse.data);
-      });
-    });
+    const fetchAdvocates = async (): Promise<void> => {
+      try {
+        const response = await fetch("/api/advocates");
+        type ApiResponse = { data: Advocate[] };
+
+        const data: ApiResponse = await response.json();
+        setAdvocates(data.data);
+        setFilteredAdvocates(data.data);
+      } catch (error) {
+        setError(error.message);
+      } finally {
+        setLoading(false);
+      }
+    };
+    fetchAdvocates();
   }, []);
 
-  const onChange = (e) => {
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const searchTerm = e.target.value;
-
-    document.getElementById("search-term").innerHTML = searchTerm;
+    console.log("searchTerm", searchTerm);
+    setSearchTerm(searchTerm);
 
     console.log("filtering advocates...");
+
+    const checkSpecialties = (specialties: string[]) => {
+      return specialties.some((specialty) =>
+        specialty.toLowerCase().includes(searchTerm.toLowerCase())
+      );
+    };
+
     const filteredAdvocates = advocates.filter((advocate) => {
+      // console.log("advocate", advocate.specialties);
       return (
-        advocate.firstName.includes(searchTerm) ||
-        advocate.lastName.includes(searchTerm) ||
-        advocate.city.includes(searchTerm) ||
-        advocate.degree.includes(searchTerm) ||
-        advocate.specialties.includes(searchTerm) ||
-        advocate.yearsOfExperience.includes(searchTerm)
+        advocate.firstName.toLowerCase().includes(searchTerm.toLowerCase()) ||
+        advocate.lastName.toLowerCase().includes(searchTerm.toLowerCase()) ||
+        advocate.city.toLowerCase().includes(searchTerm.toLowerCase()) ||
+        advocate.degree.toLowerCase().includes(searchTerm.toLowerCase()) ||
+        checkSpecialties(advocate.specialties) ||
+        advocate.yearsOfExperience.toString().includes(searchTerm)
```

Comment:
Fixes: #7 Case Sensitive Search

Previously, the filtering logic directly checked if the searchTerm was included in various advocate fields (e.g., advocate.firstName, advocate.lastName, etc.) without considering case differences. In the updated code, all string comparisons (e.g., for firstName, lastName, city, degree) are converted to lowercase using .toLowerCase() before performing the includes() check. Similarly, the searchTerm is also converted to lowercase. This ensures that the search is case-insensitive.

Users can now search for advocates without worrying about the case of their input or the case in the data.
This improves user experience by making the search functionality more flexible and intuitive.